xdsl/ir.py

```diff
@@ -781,6 +761,18 @@ def verify(self, verify_nested_ops: bool = True) -> None:
             if isinstance(operand, ErasedSSAValue):
                 raise Exception("Erased SSA value is used by the operation")
 
+        if (parent_block := self.parent) and (parent_region := parent_block.parent):
+            # TODO single-block regions dealt when the NoTerminator trait is implemented
```

Comment:
issue?

Reply:
There is no issue per se, it is just that the `IsTerminator` and `NoTerminator` checks overlap, so adding each piecewise does not allow (based on the discussion in #1049